### PR TITLE
[Merged by Bors] - Add Paddle Runtime for Adlik Serving

### PR DESCRIPTION
**Pull request by 94Bo**
Add Paddle Runtime for Adlik Serving

### CODE REVIEW

**Comment by shiyingjin** [comment]
Should add ci check

File: third_party/paddle/BUILD [hunk 1]
```diff
@@ -0,0 +1,832 @@
+load("@rules_cc//cc:defs.bzl", "cc_import", "cc_library")
+
+package(
+    default_visibility = ["//visibility:public"],
+)
+
+output_files = [
+    "paddle/include/paddle_inference_api.h",
```
Comment (shiyingjin) [inline third_party/paddle/BUILD:8]: Can these files lists be replaced by fileglob???

**Review by shiyingjin — CHANGES_REQUESTED** [review]

File: third_party/paddle/BUILD (hunk 1 above)
Comment (94Bo) [reply third_party/paddle/BUILD:8]: Glob only searches files in its own package, and looks only for source files (**not generated files** nor other targets).

File: adlik_serving/runtime/paddle/model/paddle_util.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#include "adlik_serving/framework/domain/model_config.pb.h"
+#include "paddle/include/experimental/ext_dtype.h"
+#include "paddle/include/paddle_tensor.h"
+#include "tensorflow/core/lib/core/status.h"
+
+namespace paddle_runtime {
+size_t GetDataTypeSize(const tensorflow::DataType);
+paddle::DataType ConvertToOpenVinoDataType(const tensorflow::DataType dtype);
+void CopyFromCpu(std::unique_ptr<paddle_infer::Tensor> tensor, void* buffer, const tensorflow::DataType dtype);
+void CopyToCpu(std::unique_ptr<paddle_infer::Tensor> tensor, void* buffer, const tensorflow::DataType dtype);
```
[the remark this answers is not in the document]
Comment (94Bo) [reply adlik_serving/runtime/paddle/model/paddle_util.h:19]: Paddle inference Tensor doesn't have const method `CopyToCpu`.

File: adlik_serving/runtime/paddle/model/paddle_model.cc [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+          }
+          size_t offsetByteSize = actualByteSizePerSample * offsetSize;
+          if (offsetByteSize + totalByteSize > input.byte_size) {
+            status = tensorflow::errors::InvalidArgument("unexpected size ",
+                                                         offsetByteSize + totalByteSize,
+                                                         " biggger than input blob space, expecting ",
+                                                         input.byte_size);
+            return false;
+          }
+          input.shape[0] = input.shape[0] + batchSize;
```
Comment (shiyingjin) [inline adlik_serving/runtime/paddle/model/paddle_model.cc:166]: Replace with +=

File: adlik_serving/runtime/util/datatype_size.h [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+#include "tensorflow/core/framework/types.pb.h"
+
+namespace adlik {
+namespace serving {
+
+::std::size_t GetDataTypeSize(const tensorflow::DataType dtype);
+
+}  // namespace serving
+}  // namespace adlik
+#endif
```
Comment (shiyingjin) [inline adlik_serving/runtime/util/datatype_size.h:16]: Add a new line at the end of file

File: adlik_serving/runtime/util/datatype_size.cc [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      return 8;
+    case tensorflow::DataType::DT_COMPLEX128:
+      return 16;
+    default:
+      return 0;
+  }
+}
+
+}  // namespace serving
+}  // namespace adlik
```
Comment (shiyingjin) [inline adlik_serving/runtime/util/datatype_size.cc:41]: Add a new line at the end of file

File: adlik_serving/runtime/util/datatype_size.cc [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+#include "adlik_serving/runtime/util/datatype_size.h"
+
+#include <cstddef>
+
+namespace adlik {
+namespace serving {
+
+::std::size_t GetDataTypeSize(const tensorflow::DataType dtype) {
```
Comment (shiyingjin) [inline adlik_serving/runtime/util/datatype_size.cc:11]: Maybe can use map for simplication

File: adlik_serving/runtime/paddle/model/paddle_model.cc [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+  for (auto& output : outputs) {
+    size_t offsetByteSize = 0;
+
+    for (int i = 0; i < batch.num_tasks(); ++i) {
+      adlik::serving::DimsList dims;
+      if (!ConvertDims(output.shape, dims)) {
+        INFO_LOG << "Invalid blob dims " << output.name << std::endl;
+      }
+      const BatchingMessageTask& task = batch.task(i);
+      const PredictRequestProvider* requestProvider = task.request;
```
Comment (shiyingjin) [inline adlik_serving/runtime/paddle/model/paddle_model.cc:199]: may use auto

**Pull request closed without merging**